internal/attestations/attestations.go

```diff
@@ -12,11 +14,18 @@ import (
 )
 
 const (
-	Ref                                        = "refs/gittuf/attestations"
-	referenceAuthorizationsTreeEntryName       = "reference-authorizations"
+	Ref = "refs/gittuf/attestations"
+
+	referenceAuthorizationsTreeEntryName = "reference-authorizations"
+
 	githubPullRequestAttestationsTreeEntryName = "github-pull-requests"
-	initialCommitMessage                       = "Initial commit"
-	defaultCommitMessage                       = "Update attestations"
+	githubPullRequestApprovalSystemName        = "github"
+
+	codeReviewApprovalAttestationsTreeEntryName = "code-review-approvals"
+	codeReviewApprovalIndexTreeEntryName        = "review-index.json"
```

Comment:
I was trying to keep the reference authorizations (signed by reviewer) and platform reviews (signed by platform / app on platform) separate, but we could consolidate into one tree. Above, we're recording the platform at the end of the code review path. Should we use `gittuf` for the reference authorizations and consolidate? That also addresses the question above about whether we need to record the provider or not, as I can definitely see gittuf auth and forge approvals used together, even if two different forge provider approvals is rare.

Reply:
Yeah the situation I'm hoping to avoid is needed to guess which ref is the right place to place attestations.

The question I'd pose is what value do we get from keeping these separate? In many ways if you're using an app for delegate authorizations that IS the authorization, no different than if a human reviewer directly authorized it. If you're not relying on these directly and instead using it for supplemental information, than that seems similar to if someone approves a review but isn't authorized to be authoritative (e.g. if you don't have OWNERS).

I think my expectation is that these would be stored in the same location rather than needing to search N different ref spaces to find all potential authorizations.
That said, I do think it makes sense to have separate refs for things in the repo vs things being proposed to be added to the repo - we may want to take more liberties to limit history depth to keep the repo history clean for things that aren't necessarily trusted.

Reply:
To clarify a couple things, we're using only one ref for attestations, and currently placing user-signed attestations (reference authorizations) and app-signed-on-behalf-of-scp attestations in two directories. The former is stored as a single blob (<ref>/<from>-<to> is the path in the tree, points to a blob object) while the latter is a directory in which each app can have an entry (<ref>/<from>-<to>/<scp>). I'm a little unclear on the "things being proposed to be added" bit here: do you mean attestations for a change that hasn't been merged yet are stored separately from attestations for changes that have been merged in?